### ISSUE TITLE
Support `netX.0-maccatalyst` target framework

### ISSUE THREAD

**Issue by JaneySprings** [AvaloniaUI/Avalonia#16212]
### Is your feature request related to a problem? Please describe.

Hi! I'm a creator of [.NET Meteor](https://marketplace.visualstudio.com/items?itemName=nromanov.dotnet-meteor) extension for VSCode. I want to add a Avalonia support for it. Now it already works with `iOS` and `Android` platforms (Debugging + Profiling):

<img width="1543" alt="Screenshot 2024-07-03 at 11 01 20" src="https://github.com/AvaloniaUI/Avalonia/assets/48021947/cd891b02-aaf5-4f9f-a82d-2a9c90158176">

<img width="1544" alt="Screenshot 2024-07-03 at 11 07 29" src="https://github.com/AvaloniaUI/Avalonia/assets/48021947/998146d2-334a-43aa-9b76-3386539dbb27">

But the Desktop application is not working right now. The `netX.0-macos` target framework, as I understand it, does not contain Mono libraries. I changed it to `net8.0-maccatalyst` but it threw an InvalidOperationException:
<img width="1124" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/48021947/a79efb19-5381-44d8-bb22-a4f8c71ddeea">

### Describe the solution you'd like

No InvalidOperationException if you use the `netX.0-maccatalyst` target framework.

### Describe alternatives you've considered

_

**Comment by maxkatz6** [comment]
@JaneySprings maccatalyst is basically a mobile target. It might work if you use `Avalonia.iOS` as well as running it from the same template as iOS.

Last time I checked, there was a blocked in Skia, which didn't support MacCatalyst well until newer version available in SkiaSharp 3.0. You can install it and use as described here https://github.com/AvaloniaUI/Avalonia/issues/15503

There was an idea to add "netX.0-maccatalyst" target framework to the `Avalonia.iOS` project as well.

**Comment by maxkatz6** [comment]
> The netX.0-macos target framework, as I understand it, does not contain Mono libraries.

No, it doesn't. But it should be runnable/debuggable as a normal .NET app, i.e. `dotnet run` works fine.

**Comment by JaneySprings** [comment]
> But it should be runnable/debuggable as a normal .NET app

Yes, it debuggable only with `vsdbg`. [Mono Debugger](https://github.com/mono/debugger-libs) doesn't connected to the application. 

I compared both output directories for different target frameworks, you may notice that there are no `libmono-component-*` libraries for `netX.0-macos`:

<img width="900" alt="Screenshot 2024-07-03 at 12 27 15" src="https://github.com/AvaloniaUI/Avalonia/assets/48021947/700dfdc6-f59b-436e-a47f-bcdae65749cb">

**Comment by maxkatz6** [comment]
@JaneySprings yes, `netX.0-macos` apps don't rely on mono runtime at all. So, it's rather a feature and not a bug.
The same soon might be the case with iOS as well, [as they are adopting NativeAOT and CoreCLR runtime](https://github.com/xamarin/xamarin-macios/blob/main/docs/nativeaot.md).

**Comment by JaneySprings** [comment]
Thank you for your response! I will need to explore how to work with `vsdbg` and be prepared that Microsoft will drop Mono SDB support.
